src/main/java/org/littleshoot/proxy/MitmManagerFactory.java

```diff
@@ -0,0 +1,14 @@
+package org.littleshoot.proxy;
+
+import io.netty.channel.Channel;
+
+public interface MitmManagerFactory {
+
+  /**
+   * Retrieves an instance of {@link MitmManager} based on specific attributes from cxt channel
+   *
+   * @param channel current channel from cxt
+   * @return concrete instance of {@link MitmManager}
+   */
+  MitmManager getInstance(Channel channel);
+}
```

Comment:
I do not really like exposing the whole `channel` here but there is no easy way to get all the attributes at once from the channel